walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -270,5 +281,100 @@ public void notEnoughRoom() {
+    Collection<Event> events = Arrays.asList(
+        new Event("Event 1", TimeRange.fromStartDuration(TIME_0800AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_A)),
+        new Event("Event 2", TimeRange.fromStartDuration(TIME_0900AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_B)),
+        new Event("Event 3", TimeRange.fromStartDuration(TimeRange.START_OF_DAY, TimeRange.WHOLE_DAY.duration()),
+            Arrays.asList(OPTIONAL_C))
+        );
+
+    MeetingRequest request =
+        new MeetingRequest(Arrays.asList(PERSON_A, PERSON_B), DURATION_30_MINUTES);
+    request.addOptionalAttendee(OPTIONAL_C);    
+
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected =
+        Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0800AM, false),
+            TimeRange.fromStartEnd(TIME_0830AM, TIME_0900AM, false), 
+            TimeRange.fromStartEnd(TIME_0930AM, TimeRange.END_OF_DAY, true));
+
+    Assert.assertEquals(expected, actual);      
+  }
+
+  @Test
+  public void considerNoMandatoryAttendees() {
+    // Tests for the case where there are no mandatory attendeees, but there are some 
+    // optional attendees with available slots.
+    Collection<Event> events = Arrays.asList(
+        new Event("Event 3", TimeRange.fromStartEnd(TIME_0830AM, TIME_0845AM, false),
+            Arrays.asList(OPTIONAL_B)),
+        new Event("Event 4", TimeRange.fromStartEnd(TIME_1000AM, TIME_1100AM, false),
+            Arrays.asList(OPTIONAL_A))
+        );
+
+    MeetingRequest request = new MeetingRequest(Arrays.asList(), DURATION_30_MINUTES);
+    request.addOptionalAttendee(OPTIONAL_A);
+    request.addOptionalAttendee(OPTIONAL_B);
+
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected =
+        Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0830AM, false),
+            TimeRange.fromStartEnd(TIME_0845AM, TIME_1000AM, false),
+            TimeRange.fromStartEnd(TIME_1100AM, TimeRange.END_OF_DAY, true)
+        );
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void considerOptionalAttendeesWithNoSlots() {
+      // Tests for case where there are no required attendees, and the optional attendees do not have
+      // slots available as options.
+      Collection<Event> events = Arrays.asList(
+        new Event("Event 2", TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TimeRange.END_OF_DAY, true),
+            Arrays.asList(OPTIONAL_A)),
+        new Event("Event 2", TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TimeRange.END_OF_DAY, true),
+            Arrays.asList(OPTIONAL_B))
+      );
+
+      MeetingRequest request = new MeetingRequest(Arrays.asList(), DURATION_1_HOUR);
+      request.addOptionalAttendee(OPTIONAL_A);
+      request.addOptionalAttendee(OPTIONAL_B);
+
+      Collection<TimeRange> actual = query.query(events, request);
```

Comment:
Is this test complete or not?

Reply:
now complete. i have added the assert function, and the test passed